src/block/block_ckpt.c

```diff
@@ -662,21 +662,26 @@ __ckpt_add_blkmod_entry(
   WT_SESSION_IMPL *session, WT_BLOCK_MODS *blk_mod, wt_off_t offset, wt_off_t len)
 {
     uint64_t end, start;
-    uint32_t end_rdup;
+    uint32_t end_buf_bytes, end_rdup_bytes;
 
     WT_ASSERT(session, blk_mod->granularity != 0);
     start = (uint64_t)offset / blk_mod->granularity;
     end = (uint64_t)(offset + len) / blk_mod->granularity;
     WT_ASSERT(session, end < UINT32_MAX);
-    end_rdup = WT_MAX(__wt_rduppo2((uint32_t)end, 8), WT_BLOCK_MODS_LIST_MIN);
-    if ((end_rdup << 3) > blk_mod->nbits) {
+    end_rdup_bytes = WT_MAX(__wt_rduppo2((uint32_t)end, 8), WT_BLOCK_MODS_LIST_MIN);
+    end_buf_bytes = (uint32_t)blk_mod->nbits >> 3;
```

Comment:
I don't know if `blk_mod->nbits` can be a fractional byte, but if for example it is 5, then right shift will make `end_buf_bytes` to be 0, instead of 1.  If it must be a multiple of 8 it needs an assert (and/or comment?).  I guess if it is off by one, you end up taking the `if` and maybe clearing the last byte?  It kind of looks like nbits has to be a multiple of 8 (??)

Reply:
Yes, the `end_rdup_butyes` setting a line above guarantees it is a multiple of 8. I can assert that and add a comment. (It is guaranteed as long as `WT_BLOCK_MODS_LIST_MIN` is a multiple of 8.)

Reply:
Maybe also assert `WT_BLOCK_MODS_LIST_MIN` is a multiple of 8

Reply:
Ignore that last suggestion - your assert covers it.